clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/InstanceHealthMonitor.java

```diff
@@ -26,33 +27,40 @@
 
 
 /**
- * The class is used to measure the healthiness about the cluster the store belongs to.
- * So far, it is per store because of the following reasons:
- * 1. Simplify the logic in this class since we don't need to maintain the per-store status, such as quota related responses.
- * 2. Isolate the healthiness decision among different stores to reduce the impact of false signal.
- * There are concerns with this approach as well, for example, the mis-behaving instances will take a longer time to be discovered
- * in each store.
- *
- * This class is using the pending requests + response status of each Route to decide the healthiness.
- * 1. For the good response, the pending request counter will be reset when receiving the response.
- * 2. For the error response, the pending request counter reset will be delayed, which is a way to downgrade the instance.
- * 3. When the pending request counter exceeds the pre-defined threshold, the instance will be completely blocked.
- *
+ * This class is used to track the health of the instances Fast Client is talking to.
+ * Here is the strategy:
+ * 1. It is maintaining a pending request map to track the pending request count per instance.
+ * 2. If the pending request count exceeds the configured threshold, the instance will be blocked.
+ * 3. If the request is timed out based on the configured timeout threshold, this instance will be put
+ *    into suspicious instance set.
+ * 4. The embedded heart-beat runnable will send heartbeat request to the suspicious/unhealthy instances
+ *    periodically, and if the heartbeat request still times out or receives error response, the instance
+ *    will be put into the unhealthy instance set. Otherwise, the instance will be removed from the suspicious
+ *    and unhealthy instance set.
+ * 5. Fast Client won't send user requests to the blocked instances and unhealthy instances.
  */
 public class InstanceHealthMonitor implements Closeable {
   private static final Logger LOGGER = LogManager.getLogger(InstanceHealthMonitor.class);
-  private final ClientConfig clientConfig;
+  private final InstanceHealthMonitorConfig config;
 
-  // Map/set of per store replica instances
   private final Map<String, Integer> pendingRequestCounterMap = new VeniceConcurrentHashMap<>();
   private final Set<String> unhealthyInstanceSet = new ConcurrentSkipListSet<>();
+  private final Set<String> suspiciousInstanceSet = new ConcurrentSkipListSet<>();
```

Comment:
Can you help me understand the functional difference between suspicious and unhealthy? (still reading the class, but from the javadoc I'm not sure it's apparant to me).

Reply:
For suspicious instances, fast-client can still send request to them, but the internal HB runnable will start sending HB request to these instances, and if the HB request fails, these suspicious instances will become unhealthy instances and fast-client will stop sending regular requests to them until HB requests succeed later on.